rnd/autogpt_builder/src/components/CustomNode.tsx

```diff
@@ -323,20 +324,22 @@ const CustomNode: FC<NodeProps<CustomNodeData>> = ({ data, id }) => {
         <div className="flex gap-[5px]">
           {isHovered && (
             <>
-              <button
-                className="node-action-button"
+              <Button
+                variant="outline"
+                size="icon"
                 onClick={copyNode}
                 title="Copy node"
               >
                 <Copy size={18} />
```

Comment:
Let's try removing the size from the lucide icon here. That way we can leverage our Button's size and variant without conflicts from the lucide icon.

Reply:
The lucide icons assume maximum size if it isn't specified explicitly. Radix icons don't do that and generally require little extra styling when used with shad/cn components, which is why we preferably use those. I didn't want to go as far as replacing all the icons, so this is an intermediary measure to keep the UI looking nice.

Reply:
I appreciate your attention to detail, @Pwuts . You're right that Lucide icons do fill their container by default. However, this behavior actually works in our favor when using them with ShadCN's Button component. Here's why:

The `Button` component already has built-in styles that set appropriate size constraints, including for icons when using `size="icon"`.

Letting the Button component manage icon sizing maintains visual consistency throughout our UI. By Omitting explicit sizes on icons reduces code complexity and makes future updates easier. Since ShadCN's components are designed to work harmoniously together. By not specifying icon sizes, we're leveraging this design intention.

While I understand the desire to ensure the UI looks nice, specifying sizes on the icons might actually override the implemented designed behavior of our UI components. This could lead to inconsistencies or unexpected behavior, especially if we update our design system in the future. ( Which from the discord is for sure the case )